# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/URLQueryStringStripper.h
# Commit: f1e9b8e21d42
# Full Hash: f1e9b8e21d420603b84e0aa6d043ed46e1048e44
# Author: abhishekmadan30 <148475204+abhishekmadan30@users.noreply.github.com>
# Date: 2024-03-28 21:36:34
# Regressor Bug: 1877419
# File Overlap Count: 1
# Description:
#   Bug 1877419 - Part 1: Added nested stripping of urls to copy without site tracking. r=pbz,necko-reviewers
#   
#   Depends on D204914
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202943
# ==============================================================================

diff -r ec5346caf598 -r f1e9b8e21d42 toolkit/components/antitracking/URLQueryStringStripper.h
--- a/toolkit/components/antitracking/URLQueryStringStripper.h	Thu Mar 28 18:02:16 2024 +0000
+++ b/toolkit/components/antitracking/URLQueryStringStripper.h	Thu Mar 28 18:02:16 2024 +0000
@@ -49,6 +49,12 @@
   void PopulateStripList(const nsAString& aList);
   void PopulateAllowList(const nsACString& aList);
 
+  // Recursive helper function that helps strip URIs of tracking parameters
+  // and enables the stripping of tracking paramerters that are in a URI which
+  // is nested in a query parameter
+  nsresult StripForCopyOrShareInternal(nsIURI* aURI, nsIURI** strippedURI,
+                                       int& aStripCount, bool aStripNestedURIs);
+
   nsTHashSet<nsString> mList;
   nsTHashSet<nsCString> mAllowList;
   nsCOMPtr<nsIURLQueryStrippingListService> mListService;
